test/Dialect/SV/hw-memsim.mlir

```diff
@@ -182,7 +185,7 @@ hw.module.generated @FIRRTLMemTwoAlways, @FIRRTLMem( %wo_addr_0: i4, %wo_en_0: i
   }
   // COMMON-LABEL: hw.module @FIRRTLMem_1_1_0_32_16_1_1_0_1_a
   // CHECK-SAME: (%R0_addr: i4, %R0_en: i1, %R0_clk: i1, %W0_addr: i4, %W0_en: i1, %W0_clk: i1, %W0_data: i32, %W0_mask: i4) -> (R0_data: i32)
-  // CHECK-NEXT:   %[[Memory:.+]] = sv.reg  : !hw.inout<uarray<16xi32>>
+  // CHECK-NEXT:   %[[Memory:.+]] = sv.reg
```

Comment:
The types on these checks seem important.  Usually we can elide types as not interesting, but for memories, the most interesting thing is the type.  Ditto for all the other types that were removed.